token/core/fabtoken/validator.go

```diff
@@ -54,7 +125,20 @@ func (v *Validator) VerifyTokenRequest(ledger driver.Ledger, signatureProvider d
 	for _, action := range ta {
 		actions = append(actions, action)
 	}
-
+	for _, sig := range signatureProvider.Signatures() {
```

Comment:
why is this not performed by the extravalidator?

Reply:
the extravalidator is there so it would be easier to introduce different types of scripts
since this applies only to the exchange script no reason to have an extravalidator for it

Reply:
Claim is part of the exchange script and it seems it should be validated by the `exchangeScriptValidator'. I also see that we have action Signature to carry the Claim Preimage. Action is reserved for tms actions (issue, transfer and setup). Signature itself is not an action. I see that having it this way, allow for an easy lookup of preimages, but we could probably find another way, using `TransferAction.GetMetadata()` instead.

Reply:
Will be addressed in another PR